File: schemaorg-rs/src/monetary_amount_distribution.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type MonetaryAmountDistributionAdditionalTypeFieldEnum = String;
///<https://schema.org/MonetaryAmountDistribution>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct MonetaryAmountDistribution {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/percentile25>
    #[serde(rename = "percentile25")]
    #[serde_as(as = "OneOrMany<_>")]
    pub percentile25: Vec<f32>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<MonetaryAmountDistributionAdditionalTypeFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/percentile90>
    #[serde(rename = "percentile90")]
    #[serde_as(as = "OneOrMany<_>")]
    pub percentile90: Vec<f32>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<MonetaryAmountDistributionImageFieldEnum>,
    ///<https://schema.org/currency>
    #[serde(rename = "currency")]
    #[serde_as(as = "OneOrMany<_>")]
    pub currency: Vec<String>,
    ///<https://schema.org/duration>
    #[serde(rename = "duration")]
    #[serde_as(as = "OneOrMany<_>")]
    pub duration: Vec<MonetaryAmountDistributionDurationFieldEnum>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<MonetaryAmountDistributionMainEntityOfPageFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<MonetaryAmountDistributionSubjectOfFieldEnum>,
    ///<https://schema.org/percentile10>
    #[serde(rename = "percentile10")]
    #[serde_as(as = "OneOrMany<_>")]
    pub percentile10: Vec<f32>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<MonetaryAmountDistributionDescriptionFieldEnum>,
    ///<https://schema.org/median>
    #[serde(rename = "median")]
    #[serde_as(as = "OneOrMany<_>")]
    pub median: Vec<f32>,
    ///<https://schema.org/percentile75>
    #[serde(rename = "percentile75")]
    #[serde_as(as = "OneOrMany<_>")]
    pub percentile75: Vec<f32>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<MonetaryAmountDistributionIdentifierFieldEnum>,
}

File: schemaorg-ts/src/monetary_amount_distribution.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type MonetaryAmountDistributionAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/MonetaryAmountDistribution>
 **/
export interface MonetaryAmountDistribution {
  "@context": string;
  "@type": "MonetaryAmountDistribution";

  /**
   * <https://schema.org/currency>
   **/
  currency: all.OneOrMany<string>;

  /**
   * <https://schema.org/percentile25>
   **/
  percentile25: all.OneOrMany<number>;

  /**
   * <https://schema.org/percentile75>
   **/
  percentile75: all.OneOrMany<number>;

  /**
   * <https://schema.org/median>
   **/
  median: all.OneOrMany<number>;

  /**
   * <https://schema.org/percentile90>
   **/
  percentile90: all.OneOrMany<number>;

  /**
   * <https://schema.org/duration>
   **/
  duration: all.OneOrMany<all.MonetaryAmountDistributionDurationFieldEnum>;

  /**
   * <https://schema.org/percentile10>
   **/
  percentile10: all.OneOrMany<number>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.MonetaryAmountDistributionMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.MonetaryAmountDistributionIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.MonetaryAmountDistributionDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.MonetaryAmountDistributionSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.MonetaryAmountDistributionAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.MonetaryAmountDistributionImageFieldEnum>;
}
